bundles/org.openhab.binding.openthermgateway/src/main/resources/OH-INF/thing/channels.xml

```diff
@@ -322,38 +342,378 @@
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_controlsetpoint">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Control Setpoint</label>
+		<description>Control setpoint</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_servicerequest">
+		<item-type>Switch</item-type>
+		<label>Service Request</label>
+		<description>Service request</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_exhaustfanfault">
+		<item-type>Switch</item-type>
+		<label>Exhaust Fan Fault</label>
+		<description>Exhaust fan fault</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_inletfanfault">
+		<item-type>Switch</item-type>
+		<label>Inlet Fan Fault</label>
+		<description>Inlet fan fault</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_frostprotection">
+		<item-type>Switch</item-type>
+		<label>Frost Protection</label>
+		<description>Frost protection</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_faultcode">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Fault Code</label>
+		<description>Fault code</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_diagnosticcode">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Diagnostic Code</label>
+		<description>Diagnostic code</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_systemtype">
+		<item-type>Number:Dimensionless</item-type>
+		<label>System Type</label>
+		<description>System type</description>
+		<state readOnly="true">
+			<options>
+				<option value="0">central exaust</option>
+				<option value="1">heat-recovery</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="vh_bypass">
+		<item-type>Switch</item-type>
+		<label>Bypass</label>
+		<description>Bypass</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_speedcontrol">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Speed Control</label>
+		<description>Speed control</description>
+		<state readOnly="true">
+			<options>
+				<option value="0">3-speed</option>
+				<option value="1">variable</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="vh_memberid">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Member ID</label>
+		<description>Member ID</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_openthermversion">
+		<item-type>Number:Dimensionless</item-type>
+		<label>OpenTherm Version</label>
+		<description>OpenTherm version</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="vh_versiontype">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Version Type</label>
+		<description>Version type</description>
+		<state readOnly="true"/>
+	</channel-type>
```

Comment:
These seem rather porperties than channels.

Reply:
Interesting thought, haven't looked at it like that. The values are read from the incoming stream from the OTGW just as all other values, and then based on the message code the corresponding channel is updated. This is generic logic for all channels. Apart from the value being read at some point and not known upfront.. changing the generic logic for handling incoming messages to support some of them being mapped to properties instead of channels would have quite an impact. Perhaps a nice to have for future versions?

Reply:
I won't insist on changing it either, but keep in mind that changing them to properties in a later release breaks backward compatibility. So, If you change them, you should do it now.